### DISCUSSION TITLE
Fundamentos do C#: Valores padrões

### DISCUSSION THREAD

**Discussion by JWES1971** [balta-io/forum#545] [Tira Dúvidas] [answered]
Prezado(a)s,

estou fazendo o curso de fundamentos do C#. Na aula referente aos valores padrões é informado que caso não seja definido um valor para a variável, a mesma vai adotar o valor padrão. 
Contudo, ao rodar o seguinte código obtenho um erro de compilação, que só é resolvido se uso a linha comentada ao invés da declaração direta:

```
using System;
namespace MeuApp{
    class Program
    {
        static void Main(string[] args)
        {
            // Valores padroes
            int meuInt;            
            //int meuInt = default;            
            Console.WriteLine("meuInt = {0}",meuInt);
        }
    }
}

```
O erro sendo indicado pelo compilador é:
error CS0165: Uso de variável local não atribuída "meuInt"

Durante a aula me pareceu que na declaração da variável, sem a inicialização do valor da mesma, seria automaticamente atribuído o valor padrão para o inteiro.

Estou usando a versão 8.0.103 do dotnet.

Realmente é mandatório fazer a declaração como na linha comentada, ou existe alguma forma de fazer a declaração funcionar sem declarar explicitamente na inicialização que se deve usar o default?

Agradeço antecipadamente a atenção.

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @JWES1971, como vai?

Vou melhorar o PPT na próxima versão, mas a explicação é a seguinte... os default values se aplicam a membros de classes e não locais.

Variáveis locais (Dentro de um escopo, como o Main no seu caso) precisam ser definidas antes de serem consumidas.
No caso dos tipos built-in (Value type), você tem um erro de compilação e no caso dos tipos complexos (Ref) você tem um erro de objeto nulo em tempo de execução.

Ficou mais claro?

**Comment by JWES1971** [comment]
Obrigado pela explicação!
